controllers/subscriber/subscriberControllers.js

```diff
@@ -9,11 +9,39 @@ class BaseController {
 		this._name = name;
 		this._options = options;
 		this._eventName = `d2l-subscribe-${this._name}`;
-		this._subscriptionComplete = Promise.resolve();
 	}
 }
 
 class BaseSubscriber extends BaseController {
+	constructor() {
+		super(...arguments);
+
+		this._subscriptionComplete = Promise.resolve();
+		this._timeouts = new Set();
+	}
+
+	hostDisconnected() {
+		this._timeouts.forEach(timeoutId => clearTimeout(timeoutId));
+	}
+
+	_keepTrying(callback, interval, maxTime, errorPayload, elapsedTime = 0) {
+		const response = callback();
+
+		if (response) return response;
+		if (elapsedTime >= maxTime) {
+			if (this._options.onError) this._options.onError(errorPayload);
+			return response;
```

Comment:
Is returning `response` especially helpful here since we already know it's `false`-y? Would `return;` (`return undefined`) suffice?

Reply:
For the current two consumers of this function it doesn't matter, but I've tried to keep the return value unchanged in case the consumer wanted to know exactly what falsey response came out. This way the response (after awaiting `_keepTrying`) is always a direct response from the callback they invoked.

Reply:
Cool, thanks for the context!